draft-ietf-masque-h3-datagram.md

```diff
@@ -335,238 +229,50 @@ Participant mode:
-datagram format data on this stream.
-
-RESOURCE_LIMIT:
-
-: This indicates that the context was closed to save resources. The recipient
-SHOULD limit its future registration of resource-intensive contexts.
+## The DATAGRAM Capsule {#datagram-capsule}
 
-Receipt of an unknown close code MUST be treated as if the NO_ERROR code was
-present. Close codes are registered with IANA, see {{iana-close-codes}}.
-
-
-### The Datagram Capsules {#datagram-capsule}
-
-This document defines the DATAGRAM and DATAGRAM_WITH_CONTEXT capsules types,
-known collectively as the datagram capsules (see {{iana-types}} for the value
-of the capsule types). These capsules allow an endpoint to send a datagram
-frame over an HTTP stream. This is particularly useful when using a version of
-HTTP that does not support QUIC DATAGRAM frames.
+This document defines the DATAGRAM capsule type (see {{iana-types}} for the
+value of the capsule type). This capsule allows an endpoint to send an HTTP
+Datagram over an HTTP stream. This is particularly useful when using a version
+of HTTP that does not support QUIC DATAGRAM frames.
 
 ~~~
 Datagram Capsule {
-  Type (i) = DATAGRAM or DATAGRAM_WITH_CONTEXT,
+  Type (i) = DATAGRAM,
   Length (i),
-  [Context ID (i)],
   HTTP Datagram Payload (..),
 }
 ~~~
 {: #datagram-capsule-format title="DATAGRAM Capsule Format"}
 
-Context ID:
-
-: A variable-length integer indicating the context ID of the datagram (see
-{{datagram-contexts}}). This field is present in DATAGRAM_WITH_CONTEXT capsules
-but not in DATAGRAM capsules. If a DATAGRAM capsule is used on a stream where
-datagram contexts are in use, it is associated with context ID 0.
-DATAGRAM_WITH_CONTEXT capsules MUST NOT carry context ID 0 as that context ID
-is conveyed using the DATAGRAM capsule.
-
 HTTP Datagram Payload:
 
 : The payload of the datagram, whose semantics are defined by individual
 applications. Note that this field can be empty.
 
-Datagrams sent using the datagram capsule have the exact same semantics as
+Datagrams sent using the DATAGRAM capsule have the exact same semantics as
 datagrams sent in QUIC DATAGRAM frames. In particular, the restrictions on when
 it is allowed to send an HTTP Datagram and how to process them from {{format}}
-also apply to HTTP Datagrams sent and received using the datagram capsules.
+also apply to HTTP Datagrams sent and received using the DATAGRAM capsule.
 
-The datagram capsules are transparent to intermediaries, meaning that
-intermediaries MAY parse them and send datagram capsules that they did not
-receive. This allows an intermediary to reencode HTTP Datagrams as it forwards
-them: in other words, an intermediary MAY send a datagram capsule to forward an
-HTTP Datagram which was received in a QUIC DATAGRAM frame, and vice versa.
+Intermediaries MAY parse DATAGRAM capsules and MAY send DATAGRAM capsules that
+they did not receive. This allows an intermediary to reencode HTTP Datagrams as
```

Comment:
"MAY send DATAGRAM capsules that they did not receive" is problematic.  I don't think that we want intermediaries injecting their own special wisdom into the flow, even if that might be possible, it's not what we want to specifically allow.

What you are trying to say here is in the "in other words" clause: that an intermediary may translate QUIC DATAGRAM frames into HTTP DATAGRAM capsules and vice versa.

Reply:
Agreed, rephrased the text